### ISSUE TITLE
By default, MPV will automatically adapt to the screen size. It will appear when the screen size changes. How to turn off this function

### ISSUE THREAD

**Issue by 360060316** [mpv-player/mpv#12612]
By default, MPV will automatically adapt to the screen size. It will appear when the screen size changes. How to turn off this function

**Comment by 360060316** [comment]
It's no use setting this, the video size has changed
I have set it up keepaspect=no video-aspect-override=no
When the window size changes, it will automatically change the video size

**Comment by 360060316** [comment]
> `auto-window-resize=no`

It's no use setting this, the video size has changed
I have set it up keepaspect=no video-aspect-override=no
When the window size changes, it will automatically change the video size

**Comment by sunpenghao** [comment]
Well, you can't...

Last time I checked, this behavior is hard-coded into the program logic (at least for Windows it is), and there's no option to control it.

**Comment by 360060316** [comment]
> Well, you can't...
> 
> Last time I checked, this behavior is hard-coded into the program logic (at least for Windows it is), and there's no option to control it.

Where can I remove this logical code

**Comment by 360060316** [comment]
> Well, you can't...
> 
> Last time I checked, this behavior is hard-coded into the program logic (at least for Windows it is), and there's no option to control it.

I discovered this issue on an Android device

**Comment by Dudemanguy** [comment]
>He didn't change the window size, but instead changed the video size

So `--no-keepaspect` and `--no-auto-window-resize` then. No idea how it works on Android specifically

**Comment by sunpenghao** [comment]
> `--auto-window-resize=no` is absolutely supposed to stop the window from automatically resizing when the aspect ratio changes. If it doesn't, that's a bug and please send a patch to fix it in win32.

Ah my bad, missed a line of code when I was going through `w32_common.c`. `--auto-window-resize=no` is working fine.

**Comment by sunpenghao** [comment]
> > > `--auto-window-resize=no` is absolutely supposed to stop the window from automatically resizing when the aspect ratio changes. If it doesn't, that's a bug and please send a patch to fix it in win32.
> > 
> > 
> > Ah my bad, missed a line of code when I was going through `w32_common.c`. `--auto-window-resize=no` is working fine.
> 
> It's not related to setting the window, is it? He automatically set the video size to be the same size as the window when he noticed a change in the window

No it's not. I misunderstood you earlier. If what you want is for the video to not fill the window, you should use `--video-unscaled=yes` or `--video-unscaled=downscale-big`.

**Comment by 360060316** [comment]
> > It's not related to setting the window, is it? He automatically set the video size to be the same size as the window when he noticed a change in the window
> 
> No it's not. I misunderstood you earlier. If what you want is for the video to not fill the window, you should use `--video-unscaled=yes` or `--video-unscaled=downscale-big`.

This has been tried and has no effect. It will still automatically adjust the video size

**Comment by sunpenghao** [comment]
> This has been tried and has no effect. It will still automatically adjust the video size

If you have `--no-keepaspect` then remove it. It overrides `--video-unscaled`.

**Comment by 360060316** [comment]
> > This has been tried and has no effect. It will still automatically adjust the video size
> 
> If you have `--no-keepaspect` then remove it. It overrides `--video-unscaled`.
> 
> Also, do look at the manual.

The problem remains, it will trigger when I change the window size

**Comment by sunpenghao** [comment]
> The problem remains, it will trigger when I change the window size

We can help you better if you can provide the full set of your settings.

**Comment by 360060316** [comment]
> > The problem remains, it will trigger when I change the window size
> 
> We can help you better if you can provide the full set of your settings.

There are no other settings left. The problem is that once I change the SurfaceView size, MPV will immediately automatically set the video size to match the surface. I must turn off this automatic setting and hope there is a way to turn it off

**Comment by MSRhmn** [comment]
Hello, I am very new to mpv and starting to learn it. I was looking to fix mpv default window opening to custom size that I want. Whatever the video size it is, which will by default open with a window size that I have fixed in the config file. And I found my solution from the mpv manual. For me it is the 

> geometry.


This is my mpv config: 

`hwdec=vaapi
sub-font-size=48
geometry=65%x65%
`


Anyone can find his/her solution from here I guess:

[mpv window](https://mpv.io/manual/master/#window)
